File: 35-search-insert-position/35-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        low,high=0,len(nums)
        mid = -1
        while(low<=high):
            mid = (low+high)//2
            try:    
                if(nums[mid] == target):
                    print("hig")
                    return mid
                elif(nums[mid]<target):
                    low = mid+1
                else:
                    high=mid-1
            except:
                print("hi")
                return mid
        print("hello")
        return low
        

File: 35-search-insert-position/35-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int low = 0, high = nums.length;
        int mid = -1;
        while(low<=high){
            mid = (low+high)/2;
            try{
                if(nums[mid] == target){
                    return mid;
                } else if(nums[mid]<target){
                    low = mid + 1; 
                } else {
                    high = mid - 1;
                }
            } catch(Exception e) {
                return mid;
            }
        }
        return low;
        
    }
}